lib/helpers/isAsyncFunction.js

```diff
@@ -1,9 +1,16 @@
 'use strict';
 
-const asyncFunctionPrototype = Object.getPrototypeOf(async function() {});
+let asyncFunctionPrototype = null;
+// try/catch for Babel compatibility, because Babel preset-env requires
+// regenerator-runtime for async/await and we don't want to include that
+// for a simple check.
+try {
+  asyncFunctionPrototype = Object.getPrototypeOf(async function() {});
+} catch (err) {}
 
 module.exports = function isAsyncFunction(v) {
```

Comment:
we could avoid a potential performance penalty by checking the value of asyncFunctionPrototype only once.

```javascript
module.exports =  asyncFunctionPrototype !== null
? function isAsyncFunction(v) {
  return (
    typeof v === 'function' &&
    Object.getPrototypeOf(v) === asyncFunctionPrototype
  );
}
: function () { return false; };
```